falcon/asgi/ws.py

```diff
@@ -257,10 +260,18 @@ async def close(self, code: Optional[int] = None) -> None:
         if self.closed:
             return
 
+        if code in self._close_reasons:
+            reason = self._close_reasons[code]
+        elif 3100 <= code <= 3999:
+            reason = falcon.util.code_to_http_status(code - 3000)
+        else:
+            reason = ''
```

Comment:
I think in this case we could not set the reason in the event.
we can probably create the response dict before the if and assign `reason` on it if matched.

as above we would need to account for the version